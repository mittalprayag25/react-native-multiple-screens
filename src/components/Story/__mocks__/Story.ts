import {createElement} from 'react';

jest.mock('..', () => createElement('Story'));
