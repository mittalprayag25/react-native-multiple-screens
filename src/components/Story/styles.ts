import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  story: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    marginVertical: 10,
    marginHorizontal: 5,
  },
  storyHeader: {
    fontWeight: 'bold',
    fontSize: 16,
    marginHorizontal: 5,
  },
  storyAuthor: {
    marginLeft: 10,
  },
});

export default styles;
