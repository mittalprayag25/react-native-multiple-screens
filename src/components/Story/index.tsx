import React, {memo} from 'react';
import {Text, TouchableOpacity} from 'react-native';

import {Routes} from './../../constants/NavigationUtils';

import styles from './styles';

const Story = ({records, navigation}: any): JSX.Element => {
  const showDetails = (feed: any) => {
    navigation.navigate(Routes.DETAILSCREEN, {feed});
  };

  return (
    records.length > 0 &&
    records.map((item: any) => {
      return (
        <TouchableOpacity style={styles.story}>
          <Text
            testID={`${item.feed.id}-story`}
            style={styles.storyHeader}
            onPress={() => showDetails(item.feed)}>
            {item.feed.title}
          </Text>
          <Text style={styles.storyAuthor}>(${item.feed.by})</Text>
        </TouchableOpacity>
      );
    })
  );
};

export default memo(Story);
