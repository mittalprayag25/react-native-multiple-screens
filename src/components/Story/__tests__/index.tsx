import {act, fireEvent, render} from '@testing-library/react-native';
import React from 'react';

import Story from '..';

describe('Story ', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
    records: [
      {
        id: 30827210,
        feed: {
          by: 'user',
          descendants: 99,
          id: 30827210,
          kids: [
            30828021, 30827448, 30827932, 30828013, 30827621, 30827406,
            30827414, 30827390, 30827521, 30827440, 30827379, 30827508,
            30827335, 30827377, 30827636, 30827711,
          ],
          score: 140,
          time: 1648442500,
          title: 'story1',
          type: 'story',
          url: 'https://fuchsia.dev/fuchsia-src/development/build/build_workstation',
        },
      },
      {
        id: 30827211,
        feed: {
          by: 'user1',
          descendants: 99,
          id: 30827211,
          kids: [30828021],
          score: 140,
          time: 1648442500,
          title: 'story2',
          type: 'story',
          url: 'https://fuchsia.dev/fuchsia-src/development/build/build_workstation',
        },
      },
      {
        id: 30827212,
        feed: {
          by: 'user3',
          descendants: 99,
          id: 30827212,
          score: 140,
          time: 1648442500,
          title: 'story3',
          type: 'story',
          url: 'https://fuchsia.dev/fuchsia-src/development/build/build_workstation',
        },
      },
    ],
  };

  it('should match snapshot', () => {
    const tree = render(<Story {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('test Navigation', () => {
    const {getByTestId} = render(<Story {...props} />);
    const story = getByTestId('30827210-story');
    act(() => {
      fireEvent(story, 'press');
    });
  });
});
describe('Story with no records', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
    records: [],
  };

  it('should match snapshot', () => {
    const tree = render(<Story {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
