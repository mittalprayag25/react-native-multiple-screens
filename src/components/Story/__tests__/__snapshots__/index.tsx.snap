// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story  should match snapshot 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginHorizontal": 5,
        "marginVertical": 10,
        "opacity": 1,
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "marginHorizontal": 5,
        }
      }
      testID="30827210-story"
    >
      story1
    </Text>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      ($
      user
      )
    </Text>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginHorizontal": 5,
        "marginVertical": 10,
        "opacity": 1,
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "marginHorizontal": 5,
        }
      }
      testID="30827211-story"
    >
      story2
    </Text>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      ($
      user1
      )
    </Text>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginHorizontal": 5,
        "marginVertical": 10,
        "opacity": 1,
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "marginHorizontal": 5,
        }
      }
      testID="30827212-story"
    >
      story3
    </Text>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      ($
      user3
      )
    </Text>
  </View>,
]
`;

exports[`Story with no records should match snapshot 1`] = `null`;
